pytest_mock.py

```diff
@@ -180,7 +181,9 @@ def assert_wrapper(__wrapped_mock_method__, *args, **kwargs):
             actual_args, actual_kwargs = __mock_self.call_args
             assert actual_args == args[1:]
             assert actual_kwargs == kwargs
-        raise AssertionError(*e.args)
+        err = e
+    if err is not None:
+        raise AssertionError(*err.args)
```

Comment:
Just remembered that Python 3 (can't find a link now) will delete the exception value caught by `as` after leaving the `except` block; that is done to avoiding leaking the frames in the traceback value.